providers/factories/oracle/api.go

```diff
@@ -68,27 +74,41 @@ func APIQueryHandlerFactory(
 		}
 
 		requestHandler = static.NewStaticMockClient()
+	case raydium.Name:
+		apiPriceFetcher, err = raydium.NewAPIPriceFetcher(
+			marketMap,
+			cfg.API,
+			logger,
+		)
+		if err != nil {
+			return nil, err
+		}
 	default:
 		return nil, fmt.Errorf("unknown provider: %s", cfg.Name)
 	}
 	if err != nil {
 		return nil, err
 	}
 
-	// If a custom request handler is not provided, create a new default one.
-	if requestHandler == nil {
-		requestHandler, err = apihandlers.NewRequestHandlerImpl(client)
+	// if no apiPriceFetcher has been created yet, create a default REST API price fetcher.
+	if apiPriceFetcher == nil {
+		apiPriceFetcher, err = apihandlers.NewRestAPIFetcher(
+			requestHandler,
+			apiDataHandler,
+			metrics,
+			cfg.API,
+			logger,
+		)
 		if err != nil {
 			return nil, err
 		}
 	}
```

Comment:
Let's just move this logic before the `switch` statement.  Then this is just the initial value and can be overwritten if a custom one is provided.  We can remove the `apiPriceFetcher == nil` check that way as well

Reply:
we can't move the apiPriceFetcher creation to before the switch statement, as it depends on the apiDataHandler construction (this is what the switch is for) + we need that for the apiPriceFetcher